vizro-ai/tests/unit/vizro-ai/dashboard/test_pydantic_output.py

```diff
@@ -0,0 +1,28 @@
+import vizro.models as vm
+from vizro_ai.dashboard._pydantic_output import _create_message_content, _create_prompt_template, _get_pydantic_model
+
+
+def test_get_pydantic_model(component_description, fake_llm):
+    pydantic_output = _get_pydantic_model(
+        query=component_description, llm_model=fake_llm, response_model=vm.Card, df_info=None
+    )
+    assert pydantic_output.dict(exclude={"id": True}) == vm.Card(text="this is a card", href="").dict(
+        exclude={"id": True}
+    )
+
+
+def test_create_message_content_valid(query, message_output_valid):
+    message_content = _create_message_content(query=query, df_info=None)
+
+    assert message_content == message_output_valid
+
+
+def test_create_message_content_error(query, message_output_error):
+    message_content = _create_message_content(query=query, df_info=None, validation_error="ValidationError", retry=True)
+    assert message_content == message_output_error
+
+
+def test_create_prompt_template():
+    additional_info = "Pay special attention to the following error: {validation_error}"
+    model = _create_prompt_template(additional_info)
+    assert additional_info in model.messages[0].prompt.template
```

Comment:
We can leave them in, but I think they are less important than the high-level function above! In general we should focus on testing all possible run paths of the higher level function which is `_get_pydantic_model`